### PR TITLE
ci: handle retriggering ci and human authors in `update-flake` 

### PR DESCRIPTION
**Pull request by matifali**
This PR improves #14046 based on instructions [here](https://github.com/stefanzweifel/git-auto-commit-action#commits-made-by-this-action-do-not-trigger-new-workflow-runs).
Should solve the CI stuck issue exhibited in
- #14039
- #14040 
- #14041
- #14044

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -121,25 +121,30 @@ jobs:
...
     runs-on: ${{ github.repository_owner == 'coder' && 'depot-ubuntu-22.04-8' || 'ubuntu-latest' }}
-    permissions:
-      # Give the default GITHUB_TOKEN write permission to commit and push the changed files back to the repository.
-      contents: write
     steps:
       - name: Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 1
+          token: ${{ secrets.CDRCI_GITHUB_TOKEN }}
```
Comment (matifali) [inline .github/workflows/ci.yaml:129]: This secret is already available in Secrets. I assume that it has push/write access to the repo. We need to add/update a secret if it does not work.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -121,26 +121,31 @@ jobs:
...
 
       - name: Setup Go
         uses: ./.github/actions/setup-go
 
       - name: Update Nix Flake SRI Hash
         run: ./scripts/update-flake.sh
 
+        # auto update flake for dependabot
```
Comment (deansheather) [inline .github/workflows/ci.yaml:137]:
```suggestion
      # auto update flake for dependabot
```

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -121,26 +121,31 @@ jobs:
...
       - uses: stefanzweifel/git-auto-commit-action@v5
+        if: github.actor == 'dependabot[bot]'
         with:
           # Allows dependabot to still rebase!
           commit_message: "[dependabot skip] Update Nix Flake SRI Hash"
 
+        # check for other PRs
```
Comment (deansheather) [inline .github/workflows/ci.yaml:144]:
```suggestion
      # require everyone else to update it themselves
```

**Review by deansheather — APPROVED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (deansheather) [reply .github/workflows/ci.yaml:129]: This is a PAT for the cdrci account with write perms, so it should be fine writing to PRs on this repo.

**Pull request merged**